Automattic/jetpack
# Add support for products on My Plan page and update layout to match mocks

This PR adds a support for products to the My Plan page and adjusts the page header area layout to match the new designs.

**Please note that this PR doesn't add the action buttons like "Manage Plan" or "Manage Purchase". They need to be added in a follow-up PR.**

**Before**
![Screenshot 2019-11-25 at 18 25 20](https://user-images.githubusercontent.com/478735/69563220-fd816d80-0fb0-11ea-8c0c-2ef913eb908e.png)

**After**
![Screenshot 2019-11-25 at 18 24 12](https://user-images.githubusercontent.com/478735/69563167-d9be2780-0fb0-11ea-8ed1-07e13704236d.png)


Fixes n/a

#### Changes proposed in this Pull Request:

* Migrate `MyPlanCard` from Calypso
* Refactor `MyPlanHeader` so that it supports both plans and products
* Update `MyPlanHeader` layout to match the latest mock-ups

#### Testing instructions:






* Go to /wp-admin/admin.php?page=jetpack#/my-plan
* Check if each plan information is displayed there (free and paid plans)
* Check if both Jetpack Backup information is displayed there along with the plan information. 

#### Proposed changelog entry for your changes:

* Add `MyPlanCard` card
* Refactor `MyPlanHeader` so that it supports both plans and products
